drivers/gpio/gpio_nrfx.c

```diff
@@ -253,6 +255,91 @@ static int gpio_nrfx_read(struct device *port, int access_op,
 	return 0;
 }
 
+static int gpio_nrfx_port_get_raw(struct device *port, u32_t *value)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+
+	*value = nrf_gpio_port_in_read(reg);
+
+	return 0;
+}
+
+static int gpio_nrfx_port_set_masked_raw(struct device *port, u32_t mask,
+					 u32_t value)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+	u32_t value_tmp;
+
+	value_tmp = nrf_gpio_port_out_read(reg) & ~mask;
+	nrf_gpio_port_out_write(reg, value_tmp | (mask & value));
+
+	return 0;
+}
+
+static int gpio_nrfx_port_set_bits_raw(struct device *port, u32_t mask)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+
+	nrf_gpio_port_out_set(reg, mask);
+
+	return 0;
+}
+
+static int gpio_nrfx_port_clear_bits_raw(struct device *port, u32_t mask)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+
+	nrf_gpio_port_out_clear(reg, mask);
+
+	return 0;
+}
+
+static int gpio_nrfx_port_toggle_bits(struct device *port, u32_t mask)
+{
+	NRF_GPIO_Type *reg = get_port_cfg(port)->port;
+	u32_t value;
+
+	value = nrf_gpio_port_out_read(reg);
+	nrf_gpio_port_out_write(reg, value ^ mask);
+
+	return 0;
+}
+
+static int gpio_nrfx_pin_interrupt_configure(struct device *port,
+		unsigned int pin, unsigned int flags)
+{
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	if (((flags & GPIO_INT_ENABLE) != 0) &&
+	    ((flags & GPIO_INT_EDGE) != 0) &&
+	    (nrf_gpio_pin_dir_get(
+		NRF_GPIO_PIN_MAP(get_port_cfg(port)->port_num, pin)) != 0)) {
+		/*
+		 * The pin's output value as specified in the GPIO will be
+		 * ignored as long as the pin is controlled by GPIOTE.
+		 * Attempting to write a pin as a normal GPIO pin will have no
+		 * effect. Controlling the pin via GPIOTE, used to trigger edge
+		 * interrupts, is currently not supported.
```

Comment:
I think I don't understand the last sentence. What is not supported is configuring the edge triggered interrupt for a pin configured as an output. And that is because for triggering such interrupt we use a GPIOTE channel in Event mode, what effectively configures the pin as an input, overriding the DIR setting in GPIO.